### PR TITLE
Improve the Job API comment for ready field

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind documentation
/kind api-change

#### What this PR does / why we need it:

To reflect the status quo that pods with deletionTimestamp are not counted. The JobPodsReady KEP is already GA so I assume this is the expect behavior, even if under-specified (see [here](https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/2879-ready-pods-job-status/README.md)).

The pods with deletion timestamp are excluded because ready is computed based on the active pods (see [here](https://github.com/kubernetes/kubernetes/blob/afebfdc5d411859468153e2b9921915f3e4f9ff4/pkg/controller/job/job_controller.go#L832)). And the active pods are filtered with exclusion of terminating pods, see the definition of active pod [here](https://github.com/kubernetes/kubernetes/blob/afebfdc5d411859468153e2b9921915f3e4f9ff4/pkg/controller/controller_utils.go#L980-L984)

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/2879-ready-pods-job-status/README.md
```

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve
/label api-review

File: staging/src/k8s.io/api/batch/v1/types.go [hunk 1]
```diff
@@ -595,8 +595,8 @@ type JobStatus struct {
 	// +optional
 	UncountedTerminatedPods *UncountedTerminatedPods `json:"uncountedTerminatedPods,omitempty" protobuf:"bytes,8,opt,name=uncountedTerminatedPods"`
 
-	// The number of pods which have a Ready condition.
-	// +optional
```
Comment (deads2k) [inline staging/src/k8s.io/api/batch/v1/types.go:599]: why did optional get removed?

**Pull request merged**